### PR TITLE
Onboarding keyvault for autogeneration

### PR DESCRIPTION
**Pull request by qinl-li**

### CODE REVIEW

**Comment by qinl-li** [comment]
> azure-schema-generator@1.0.0 generate-single C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "keyvault/resource-manager"

[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\tojin\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
[C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=C:\Users\tojin\AppData\Local\Temp\owxrhfpz36i --tag=all-api-versions --api-version=2018-02-14-preview --title=none --pass-thru:subset-reducer C:\Users\tojin\AppData\Local\Temp\schm_azspc\specification\keyvault\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\tojin\.autorest\@autorest_core@3.0.6257\node_modules\@autorest\core\dist' (3.0.6257)
   Installing AutoRest extension '@autorest/azureresourceschema' (3.0.79)
   Installed AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[11.45 s] Generation Complete
================================================================================================================================
Filename: C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\schemas\2018-02-14-preview\Microsoft.KeyVault.json
Provider Namespace: Microsoft.KeyVault
API Version: 2018-02-14-preview
Resource Types (Resource Group Scope):
- vaults
- vaults/accessPolicies
================================================================================================================================
[C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=C:\Users\tojin\AppData\Local\Temp\p8w5gxy9rvf --tag=all-api-versions --api-version=2015-06-01 --title=none --pass-thru:subset-reducer C:\Users\tojin\AppData\Local\Temp\schm_azspc\specification\keyvault\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\tojin\.autorest\@autorest_core@3.0.6257\node_modules\@autorest\core\dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[7.43 s] Generation Complete
================================================================================================================================
Filename: C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\schemas\2015-06-01\Microsoft.KeyVault.json
Provider Namespace: Microsoft.KeyVault
API Version: 2015-06-01
Resource Types (Resource Group Scope):
- vaults
================================================================================================================================
[C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=C:\Users\tojin\AppData\Local\Temp\plbw3cmfgvd --tag=all-api-versions --api-version=2016-10-01 --title=none --pass-thru:subset-reducer C:\Users\tojin\AppData\Local\Temp\schm_azspc\specification\keyvault\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\tojin\.autorest\@autorest_core@3.0.6257\node_modules\@autorest\core\dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[8.21 s] Generation Complete
================================================================================================================================
Filename: C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\schemas\2016-10-01\Microsoft.KeyVault.json
Provider Namespace: Microsoft.KeyVault
API Version: 2016-10-01
Resource Types (Resource Group Scope):
- vaults
- vaults/accessPolicies
================================================================================================================================
[C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=C:\Users\tojin\AppData\Local\Temp\hmlz5b0fzyu --tag=all-api-versions --api-version=2018-02-14 --title=none --pass-thru:subset-reducer C:\Users\tojin\AppData\Local\Temp\schm_azspc\specification\keyvault\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\tojin\.autorest\@autorest_core@3.0.6257\node_modules\@autorest\core\dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[8.82 s] Generation Complete
================================================================================================================================
Filename: C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\schemas\2018-02-14\Microsoft.KeyVault.json
Provider Namespace: Microsoft.KeyVault
API Version: 2018-02-14
Resource Types (Resource Group Scope):
- vaults
- vaults/accessPolicies
- vaults/privateEndpointConnections
================================================================================================================================
[C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6257 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=C:\Users\tojin\AppData\Local\Temp\pv4xok3ozeg --tag=all-api-versions --api-version=2019-09-01 --title=none --pass-thru:subset-reducer C:\Users\tojin\AppData\Local\Temp\schm_azspc\specification\keyvault\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\tojin\.autorest\@autorest_core@3.0.6257\node_modules\@autorest\core\dist' (3.0.6257)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[8.8 s] Generation Complete
================================================================================================================================
Filename: C:\Users\tojin\Documents\GitHub\azure-resource-manager-schemas\schemas\2019-09-01\Microsoft.KeyVault.json
Provider Namespace: Microsoft.KeyVault
API Version: 2019-09-01
Resource Types (Resource Group Scope):
- vaults
- vaults/accessPolicies
- vaults/privateEndpointConnections
================================================================================================================================

**Comment by qinl-li** [comment]
[generate schemas output.txt](https://github.com/Azure/azure-resource-manager-schemas/files/4446163/generate.schemas.output.txt)

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -67,10 +67,16 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-04-08/Microsoft.DocumentDB.json#/resourceDefinitions/databaseAccounts_apis_keyspaces_tables" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-04-08/Microsoft.DocumentDB.json#/resourceDefinitions/databaseAccounts_apis_tables" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-06-01/Microsoft.KeyVault.json#/resourceDefinitions/vaults" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-06-01/Microsoft.KeyVault.json#/resourceDefinitions/secrets" },
```
Comment (leni-msft) [inline schemas/2014-04-01-preview/deploymentTemplate.json:70]: IIRC, this resource is useful in ARM template, so ARM wants to keep it. @anthony-c-martin could you confirm?

Comment (anthony-c-martin) [reply schemas/2014-04-01-preview/deploymentTemplate.json:70]: Yup, that's correct.

**Pull request closed without merging**